src/services/control-plane/app/control-plane.ts

```diff
@@ -0,0 +1,72 @@
+import { ForRepoQuerying } from '../ports/drivens'
+import { ForMonitoringAuthDetails } from '../ports/drivens/for-monitoring'
+import { ForManagingAuthDetails } from '../ports/drivers/for-managing-auth-details'
+import { AuthDetails, Permissions } from './schemas'
+import jwt from 'jsonwebtoken'
+
+export class ControlPlane implements ForManagingAuthDetails {
+  private secretKey: string = 'mySecretKey'
+  constructor(
+    private readonly logger: ForMonitoringAuthDetails,
+    private readonly repoQuerier: ForRepoQuerying
+  ) {}
+
+  async getAuthDetails(email: string, password: string): Promise<AuthDetails> {
+    const generateToken = (
+      payload: object,
+      secretKey: string,
+      expiresIn: string
+    ): string => {
+      const token = jwt.sign(payload, secretKey, { expiresIn })
+
+      return token
+    }
+    const generateRefreshToken = (
+      payload: object,
+      secretKey: string,
+      expiresIn: string
+    ): string => {
+      const refreshToken = jwt.sign(payload, secretKey, { expiresIn })
+
+      return refreshToken
+    }
+    const token = generateToken({ email, password }, this.secretKey, '30m')
```

Comment:
Dejar espacios entre unidades lógicas
```suggestion
    const token = generateToken({ email, password }, this.secretKey, '30m')
    
```

Reply:
'30m' podría ser una variable que puede ser modificable por algún método para que sea congruente en toda la app